kernel/thread.c

```diff
@@ -726,7 +726,7 @@ bool z_spin_lock_valid(struct k_spinlock *l)
 
 bool z_spin_unlock_valid(struct k_spinlock *l)
 {
-	if (l->thread_cpu != (_current_cpu->id | (u32_t)_current)) {
+	if (l->thread_cpu != (_current_cpu->id | (uintptr_t)_current)) {
```

Comment:
Heh, egg on my face on this one.  Actually I think this code ran originally on x86_64, so it's not like I wasn't thinking about pointer size issues, heh.